ethereumj-core/src/main/java/org/ethereum/casper/core/CasperFacade.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) [2016] [ <ether.camp> ]
+ * This file is part of the ethereumJ library.
+ *
+ * The ethereumJ library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ethereumJ library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ethereumJ library. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.casper.core;
+
+import org.ethereum.casper.config.CasperProperties;
+import org.ethereum.config.SystemProperties;
+import org.ethereum.core.Block;
+import org.ethereum.core.CallTransaction;
+import org.ethereum.core.Transaction;
+import org.ethereum.facade.Ethereum;
+import org.ethereum.vm.program.ProgramResult;
+import org.spongycastle.util.encoders.Hex;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class CasperFacade {
+
+    private CasperProperties systemProperties;
+
+    private Ethereum ethereum;
+
+    private CallTransaction.Contract casper = null;
+
+    private String contractAddress;  // FIXME: why should we have casper addresses in two places?. It's already in SystemProperties
+
+    private List<Transaction> initTxs;
+
+    public CallTransaction.Contract getContract() {
+        init();
+        return casper;
+    }
+
+    private void init() {
+        if (casper == null) {
+            contractAddress = Hex.toHexString(systemProperties.getCasperAddress());
+            String casperAbi = systemProperties.getCasperAbi();
+            casper = new CallTransaction.Contract(casperAbi);
+        }
+    }
+
+    public Object[] constCall(String func, Object... funcArgs) {
+        init();
+        ProgramResult r = ethereum.callConstantFunction(contractAddress,
```

Comment:
Looks like a call to a function needs its own code abstraction marked by `@Scope("prototype")` for easy dependency resolving. Thus, client won't always need an `Ethereum` instance to call a function.

Reply:
Nice catch, will try to implement.